src/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -565,5 +565,15 @@ public override void EndWrite(IAsyncResult asyncResult)
         {
             _sslState.SecureStream.EndWrite(asyncResult);
         }
+
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
```

Comment:
We should probably do the explicit parameter validation here.  I realize ReadOnlyMemory will do it as well, but it'll do it with the wrong parameter names. 😦

Reply:
Ahh so the issue is that exception will change the name that is thrown out in the exception of invalid sizes?

Reply:
`ReadOnlyMemory` does this validation:
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/ReadOnlyMemory.cs#L60-L65
so if the `buffer` is null, it'll throw an `ArgumentNullException` but citing the parameter name `array`, and if `offset` or `count` are incorrect, it'll throw an `ArgumentOutOfRangeException`, but without any parameter name (and if `ReadOnlyMemory` is changed to include a name, it currently uses `index` and `length` instead of `offset` and `count`.)
cc: @ahsonkhan, @KrzysztofCwalina

Reply:
I guess (and that might be why you cc'd people) looking at how many times the naming

(array, index, length)
as opposed to
(buffer, offset, count)

(I suspect the second is more common)

Reply:
I added an issue here #24296

To discuss it

Reply:
We should add the explicit validation.  We can remove it later based on the outcome of #24296.

Reply:
I already did. I put the array, offset, count overload back into sslstreaminternal and the explicit validation in there as that's the traditional level for it.

Reply:
Ah, ok, thanks.